middle_end/flambda2/types/reify.ml

```diff
@@ -207,27 +211,45 @@ let reify ~allowed_if_free_vars_defined_in ~var_is_defined_at_toplevel
         if Expand_head.is_bottom env imms
         then
           match TG.Row_like_for_blocks.get_singleton blocks with
-          | None -> try_canonical_simple ()
+          | None ->
+            (* CR mshinwell: could recognise tagged immediates *)
+            try_canonical_simple ()
           | Some (tag, shape, size, field_types, alloc_mode) -> (
             assert (
               Targetint_31_63.equal size
                 (TG.Product.Int_indexed.width field_types));
-            (* CR mshinwell: Could recognise other things, e.g. tagged
-               immediates and float arrays, supported by [Static_part]. *)
+            let field_types = TG.Product.Int_indexed.components field_types in
+            let field_kinds =
```

Comment:
I think you could build the combined list directly here, instead of passing two lists to `try_to_reify_fields` which will combine them anyway.